examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -21,20 +21,116 @@
 
 #include <iostream>
 #include <list>
+#include <vector>
+
+struct CachedMessageOption
+{
+    CachedMessageOption(uint32_t id, std::string label) :
+        mLabel(label), mOption{ chip::MakeOptional(id), chip::MakeOptional(chip::CharSpan::fromCharString(mLabel.c_str())) }
+    {}
+
+    CachedMessageOption(const CachedMessageOption & option) :
+        mLabel(option.mLabel),
+        mOption{ option.mOption.messageResponseID, chip::MakeOptional(chip::CharSpan::fromCharString(mLabel.c_str())) }
+    {}
+
+    // CachedMessageOption & operator=(const CachedMessageOption & option) { return *this; };
```

Comment:
This has to be a `=delete` here, otherwise the compiler-generated operator= is invalid. This is a self-referencing object where mOption has to point into mLabel. If you copy mOption bitwise from somewhere else it will be wrong.

So either (a) implement a valid assignment or (b) make it ` = delete`.

Reply:
marked the assignment operator as delete